backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/S3BackupStore.java

```diff
@@ -343,7 +355,18 @@ private CompletableFuture<PutObjectResponse> saveNamedFile(
         AsyncRequestBody.fromFile(filePath));
   }
 
-  public static String objectPrefix(BackupIdentifier id) {
-    return "%s/%s/%s/".formatted(id.partitionId(), id.checkpointId(), id.nodeId());
+  private static S3AsyncClient buildClient(S3BackupConfig config) {
+    final var builder = S3AsyncClient.builder();
+    config.endpoint().ifPresent(endpoint -> builder.endpointOverride(URI.create(endpoint)));
+    config.region().ifPresent(region -> builder.region(Region.of(region)));
+    config
+        .credentials()
+        .ifPresent(
+            credentials ->
+                builder.credentialsProvider(
+                    StaticCredentialsProvider.create(
+                        AwsBasicCredentials.create(
+                            credentials.accessKey(), credentials.secretKey()))));
+    return builder.build();
```

Comment:
:question: What would happen if there are no credential provided and no credentials available from environment or other ways. Does the build fails? or does it fail when the first time it tries to access S3?